channels/virtual-state-channels.md

```diff
@@ -0,0 +1,498 @@
+intermediary is and the initial token amounts dedicated by the participant and
+the intermediary. An important detail is that the initial token amounts for
+the participants are not according to the amounts they've dedicatrs: it is the
+intermediary that has all of the tokens and the Virtual State Channel
+participant has none. This plays an important role and it is described later
+on.
+
+##### Example
+
+For this example we will be using some pseudo-structures. Those do not
+necessarily represent real serialization and play mostly a visualization role.
+
+If an on-chain AliceBob State Channel would look like:
+
+```javascript=
+{
+   "id": "ch_...", // the Alice-Bob channel ID
+   "initiator_id": "ak_...", // the address of Alice
+   "responder_id": "ak_...", // the address of Bob
+   "initiator_amount": 3, // Alice balance
+   "responder_amount": 2, // Bob balance
+   "channel_amount": 5 // the sum of the two amounts
+
+}
+```
+
+The Virtual Alice Bob State Channel object where Ingrid is an intermediary
+would be:
+
+```javascript=
+{
+   "intermediary_id": "ak_...", // Ingrid's address
+   "intermediary_amount": 5, // initial distribution of total tokens, dedicated to Ingrid
+   "requester_amount": 0, // initial distribution of total tokens, dedicated to Alice
+   "channel" :
+      {  
+         "id": "ch_...", // the Alice-Bob channel ID
+         "initiator_id": "ak_...", // the address of Alice
+         "responder_id": "ak_...", // the address of Bob
+         "initiator_amount": 3, // Alice balance
+         "responder_amount": 2, // Bob balance
+         "channel_amount": 5 // the sum of the two amounts
+      }
+}
+```
+
+#### Object updates
+
+The Virtual State Channel object is persisted in the state of the
+corresponding parent State Channels and can be updated by providing off-chain
+transactions that belong to the Virtual State Channel. Similar rules apply as
+with on-chain State Channels: newer state replaces old one. Off-chain updates
+contain a number `round` describing the point of time when the update had been
+made. Since it is ever-incrementing, two off-chain updates can be compared. If
+two share the same `round` and one is co-authenticated and the other had been
+produced by some unilateral action, the former takes precedence over the
+latter.
+
+Virtual State Channel participants create and co-authenticate desired updates
+and provide them to the intermediary in the respective parent State Channels.
+In order for them to be applied there, the intermediary must also agree that
+those are valid updates. If the intermediary refuses to cooperate, most of
+those can be forcfully progressed on-chain. All of those will be explained in
+detail.
+
+A special case arises when both Virtual State Channel participants agree on an
+off-chain update and one of them provides it to the intermediary while the
+other does not. Since Channel state can only be updated with a mutual
```

Comment:
But would this happen? Didn't we say that the initiator of the update sends to the intermediary, and the intermediary forwards it to the other parent channel? This would be more similar to how it is on the chain (only one party posts to the chain, while the other gets notified of the on-chain change).

Reply:
Nope, it is the other way around: VSC participants create their updates and each of them provides them to the intermediary in their parent State Channels.

The other way around actually imposes a bit of a risk for the intermediary, namely:
1) Alice and Bob agree on stateX
2) Alice gives Ingrid the stateX
3) Ingrid gives Bob stateX but Bob refuses to update their VSC object

What shall Ingrid do now? In order for her to be safe, she must include that in their channel so she shall force progress it on-chain. On the other hand, Bob is in a possession of the update, already signed by Ingrid, and question is this gives him any advantage or not. If Ingrid ever authenticates an alternative update with the same `round`, she is in trouble.

A safer approach I am slightly in favour of would be both Alice and Bob provide her with the VSC off-chain update they have for her. If just one of them does it and the other does not, Alice could gently poke the non-cooperative one with a soft-force progress action or go directly on-chain forcing it there.

Reply:
I take that back. This is part of the category of problems _A proposes something authenticated to B, B refuses but is still in possession of the authenticated object_ . The worst thing could happen is Bob to force progress based on that particular state - hence approving it. The flow who tells who what should not really matter that much and we need propagation of messages in multy intermediary scenarios anyway.

The described approach is not mandatorary but prefered for the symmetry it provides.

Reply:
But the safer approach would mean that Ingrid gets an update from Alice, and then waits for the same tx to be provided by Bob, eventually perhaps poking with a soft-FP. If Ingrid instead performs the soft-FP right away, Bob would be waiting in much the same way as in the on-chain case, and refusing would either have the result that Alice FPs on the parent chain/channel or simply solo-closes and declares the VSC uncooperative.